CMR/python/docit.py

```diff
@@ -0,0 +1,387 @@
+    """
+    Look for a local from value and make it global by prepending the module name
+    Parameters:
+        module: a python module (not a string)
+        data: dictionary of the document-it tag
+    """
+    if 'from' in data and data['from'].startswith('.'):
+        data['from'] = module.__name__ + data['from']
+
+def handle_function(func_db, module, out, obj_info, config):
+    """
+    Most of the work from tree is here, print out the documentation for a
+    function
+    Parameters:
+        func_db: function database, all the functions and their doc-it strings
+        module: module currently working on
+        out: list of markdown strings to be sent as output
+        obj_info: (object type, object)
+        config: Configuration dictionary
+    """
+    func_name = obj_info[0]
+    obj = obj_info[1]
+    first = True
+    comments = inspect.getcomments(obj)
+    if comments is not None and len(comments) > 0:
+        for raw_item in comments.strip().split("\n"):
+            posible_item = raw_item.strip()
+            if len(posible_item) < 1:
+                continue
+            if posible_item.startswith(DOCUMENT_IT):
+                item = posible_item[len(DOCUMENT_IT):]
+                if first:
+                    # Function documentation is inserted herehere
+                    out.append (FORMAT_FUNC.format(func_name))
+                    if config.get('verbose', False) and obj.__doc__ is not None:
+                        out.append (obj.__doc__)
+                    if func_name in config:
+                        out.append(config.get(func_name, '') + '\n')
+                data = json.loads(item)
+
+                if 'from' in data:
+                    fix_local_from(module, data)
+                    find_froms(func_db, out, data)
+                else:
+                    txt = document_it_tag_to_str(item)
+                    out.append ("* {}".format( txt ))
+                first = False
+    if not first:
+        out.append('')
+
+def handle_builtins(out, obj_info, config:dict):
+    """
+    Handle the built-in use case
+    Parameters:
+        out: list of strings to be sent through print() as the markdown content
+        obj_info: (object type, object)
+        config: configuration dictionary, look for 'verbose'
+    """
+    obj_type = obj_info[0]
+    obj = obj_info[1]
+    if config.get('verbose', False) and obj.__doc__ is not None:
+        out.append("* Global Dictionary - {}".format(obj_type))
+        out.append (obj.__doc__)
+
+
+# ##############################################################################
+#mark - Public Functions
+
+def pre_tree(func_db, module, config=None, cache=None):
+    """
+    Scan the modules ahead of time and make a map of all the documents-its for
+    future use. Must be called before tree(). This database of functions will
+    then be called by tree() to query and walk the 'froms'
+    """
+    config = config if isinstance(config, dict) else {}
```

Comment:
same comment about modularization of this line

Reply:
what do you suggest here?

Reply:
wrote cmr.util.common.always()